File: app/src/main/java/studio/mandysa/music/service/model/DefaultMusic.java
package studio.mandysa.music.service.model;

import java.util.ArrayList;

public class DefaultMusic implements MusicModel {

    private String mTitle;

    private String mCoverUrl;

    private String mUrl;

    private final ArrayList<DefaultArtist> mArtistList = new ArrayList<>();

    public DefaultMusic setTitle(String title) {
        mTitle = title;
        return this;
    }

    @Override
    public String getTitle() {
        return mTitle;
    }

    public DefaultMusic setCoverUrl(String url) {
        mCoverUrl = url;
        return this;
    }

    @Override
    public String getCoverUrl() {
        return mCoverUrl;
    }

    public DefaultMusic setUrl(String url) {
        mUrl = url;
        return this;
    }

    @Override
    public String getUrl() {
        return mUrl;
    }

    @Override
    public ArrayList<DefaultArtist> getArtist() {
        return mArtistList;
    }

}

File: media/src/main/java/mandysax/media/model/DefaultMusic.kt
package mandysax.media.model

import java.util.*

open class DefaultMusic<E : ArtistModel?> : MusicModel<E> {
    private val mArtistList = ArrayList<E>()
    private var mTitle: String? = null
    private var mCoverUrl: String? = null
    private var mUrl: String? = null
    private var id: String? = null

    fun setId(id: String?): DefaultMusic<E> {
        this.id = id
        return this
    }

    override fun getId() = id

    override fun getTitle(): String {
        return mTitle!!
    }

    fun setTitle(title: String?) {
        mTitle = title
    }

    override fun getCoverUrl(): String {
        return mCoverUrl!!
    }

    fun setCoverUrl(url: String?) {
        mCoverUrl = url
    }

    override fun getUrl(): String {
        return mUrl!!
    }

    fun setUrl(url: String?) {
        mUrl = url
    }

    fun addArtist(artistModel: E) {
        mArtistList.add(artistModel)
    }

    override fun getArtist(): List<E> {
        return mArtistList
    }
}